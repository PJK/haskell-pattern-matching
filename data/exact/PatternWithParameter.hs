module PatternWithParameter where

data MyTrinary = A | B | C

data MyMaybe = Sum MyTrinary | Nope

extract :: MyMaybe -> MyTrinary
extract (Sum C) = A
extract (Sum A) = B

