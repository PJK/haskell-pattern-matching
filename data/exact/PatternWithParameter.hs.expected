{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "extract",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "B",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ]
                ]
            }
        ]
    ]
}
